components/brave_wallet/browser/filecoin_keyring.cc

```diff
@@ -49,22 +51,32 @@ FilecoinKeyring::Type FilecoinKeyring::type() const {
 
 std::string FilecoinKeyring::ImportFilecoinBLSAccount(
     const std::vector<uint8_t>& private_key,
-    const std::vector<uint8_t>& public_key,
     const std::string& network) {
-  if (private_key.empty() || public_key.empty()) {
+#if defined(ARCH_CPU_64_BITS)
+  if (private_key.empty()) {
     return std::string();
   }
 
   std::unique_ptr<HDKey> hd_key = HDKey::GenerateFromPrivateKey(private_key);
   if (!hd_key)
     return std::string();
+
   int protocol = static_cast<int>(mojom::FilecoinAddressProtocol::BLS);
+  std::array<uint8_t, 32> payload;
+  std::copy_n(private_key.begin(), 32, payload.begin());
+  auto result = fil_private_key_public_key(payload);
+  if (result.empty())
```

Comment:
this would never be empty because our interface is defined as `[u8; 48]`, even `PrivateKey::from_bytes`failed it is not empty

Reply:
either change to use `[u8]` or check all zero or use a struct to return success and public key

Reply:
done

Reply:
sorry check all zero is not a good idea, that might be a valid public key

Reply:
discussed it with @dignifiedquire from filecoin team, he said we should not allow users to interact with the chain, using the all 0s key